runtime/autoload/man.vim

```diff
@@ -20,25 +22,16 @@ function man#get_page(...) abort
   elseif a:0 > 2
     echoerr 'too many arguments'
     return
-  elseif a:0 == 2
-    let [page, sect] = [a:2, 0 + a:1]
-  elseif type(1) == type(a:1)
-    let [page, sect] = ['<cword>', a:1]
-  else
-    let [page, sect] = [a:1, '']
   endif
 
-  if page == '<cword>'
-    let page = expand('<cword>')
-  endif
+  let sect = get(a:000, 0)
+  let page = get(a:000, 1, sect)
 
   let [page, sect] = s:parse_page_and_section(sect, page)
 
-  if 0 + sect > 0 && s:find_page(sect, page) == 0
+  if !empty(sect) && s:find_page(sect, page) == 0
     let sect = ''
-  endif
-
-  if s:find_page(sect, page) == 0
+  elseif s:find_page(sect, page) == 0
```

Comment:
This change will call s:find_page only once, previously it did this for the second time with empty section when section was not empty, but page with this section was not found.